### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by ashmalzahra**
### Create models
-  Create model classes for all entities as shown in the [ERD diagram](../images/blog_app_erd.png).
-  Set up associations between models.

### Use models to insert data

- Create at least one user by running the following code:
    ```ruby
       first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
       second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
    ```
- Create at least 4 posts written by one of the users you created by running the following code:
    ```ruby
       first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
    ```
- Create at least 6 posts comments for one of the posts you created by running the following code:
    ```ruby
       Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
    ```

### Create custom methods
-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, class_name: 'Post', foreign_key: 'post_id'
+  def update_like_counter
+    post.increment!(:likes_counter)
+  end
```
Comment (MaganaAsiati) [inline app/models/like.rb:10]: - Kindly use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_like_counter method every time a new like has been saved to database, then make the update_like_counter method private :+1:

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Post < ApplicationRecord
+  has_many :likes, foreign_key: 'post_id'
+  has_many :comments, foreign_key: 'post_id'
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  def update_post_counter
+    author.increment!(:posts_counter)
+  end
+
```
Comment (MaganaAsiati) [inline app/models/post.rb]: Kindly use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private :+1: 

eg.
 ```ruby
 # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, class_name: 'Post', foreign_key: 'post_id'
+  def update_comment_counter
+    post.increment!(:comments_counter)
+  end
```
Comment (MaganaAsiati) [inline app/models/comment.rb:10]: Kindly use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_comment_counter` method every time a new comment has been saved to the database, then make the `update_comment_counter` method private :+1:

**Review by MaganaAsiati — CHANGES_REQUESTED** [review]
# STATUS: Required Changes ♻️♻️♻️ 

Hi @ashmalzahra 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

AlonzoLeroneThisCloseGIF

## Highlights
✅ No linter error
✅ README is professional

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by Danieldotcomcoder — APPROVED** [review]
Hello @ashmalzahra , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 👍 👍 😄 👏 

Congratulations! 🎉 🎉 🎉 

## Highlights
- Professional Readme File ✔✔
- Linters are setup correctly ✔✔
- Clear and nice code ✔✔

comp-2
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**